qbraid/transpiler/conversions/openqasm3/openqasm3_to_cudaq.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+
+"""
+Module containing OpenQASM to CUDA-Q conversion function
+
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Union
+
+import cudaq
+from openqasm3 import ast
+import pyqasm
+
+from qbraid.transpiler.annotations import weight
+from qbraid.transpiler.exceptions import ProgramConversionError
+
+if TYPE_CHECKING:
+    from cudaq import PyKernel, QuakeValue
+
+
+def gate_kernel(name: str, *args) -> PyKernel:
+    if name in ["x", "y", "z", "h", "s", "t"]:
+        size, argc = 1, 0
+    elif name in ["rx", "ry", "rz"]:
+        size, argc = 1, 1
+    elif name in ["swap"]:
+        size, argc = 2, 0
+    elif name in ["u3"]:
+        size, argc = 3, 3
+    else:
+        raise ProgramConversionError(f"Unsupported gate: {name}")
+
+    kernel, *qrefs = cudaq.make_kernel(*[cudaq.qubit for _ in range(size)])
+
+    if len(args) != argc:
+        raise ProgramConversionError(
+            f"Gate {name} requires {argc} args but only {len(args)} were provided: {args}"
+        )
+
+    op = getattr(kernel, name)
+
+    if len(args) > 0:
+        op(*args, *qrefs)
+    else:
+        op(*qrefs)
+
+    return kernel
+
+
+def identifier_lookup(cbit: Union[ast.IndexedIdentifier, ast.Identifier]) -> str:
+    """Unpacks the identifier name"""
+    if isinstance(cbit, ast.IndexedIdentifier):
+        return cbit.name.name
+
+    return cbit.name
+
+
+@weight(1)
+def openqasm3_to_cudaq(program: ast.Program) -> PyKernel:
```

Comment:
Need the `@requires_extras("cudaq")` decorator as well (imported from `qbraid.transpiler.annotations`)

This is so that in [`ConversionGraph.create_conversion_graph`](https://github.com/qBraid/qBraid/blob/896d6fb66da6436ee6796d36d17f5ea834eb9ed9/qbraid/transpiler/graph.py#L144-L146), the `openqasm3_to_cudaq` conversion is not included if [`Conversion.supported`](https://github.com/qBraid/qBraid/blob/896d6fb66da6436ee6796d36d17f5ea834eb9ed9/qbraid/transpiler/edge.py#L99) returns `False`, indicating that `cudaq` failed to import